From 3dcd5e759f0a2f4e5b6c04e872739a7a9125a85f Mon Sep 17 00:00:00 2001
From: Conn O'Griofa <connogriofa@gmail.com>
Date: Sat, 18 Jan 2014 18:39:52 +0000
Subject: [PATCH] Implement BOARD_CAMERA_NO_UNWANTED_MSG

On msm7x27 devices, most camera hals appear to spam this message:

lockIfMessageWanted(16): dropped unwanted message

This happens for all camera modes except barcode scanning. If there
is a better fix for this issue, please let me know.

Conflicts:
	services/camera/libcameraservice/Android.mk

(cherry picked from commit 4a441201d19178082f7fc4978dc308c8ac9edf84)

Change-Id: Ie027dc73537f264e16605209646e7b62bc6e6941
---
 services/camera/libcameraservice/Android.mk       | 4 ++++
 services/camera/libcameraservice/CameraClient.cpp | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/services/camera/libcameraservice/Android.mk b/services/camera/libcameraservice/Android.mk
index d0b28d4..c2cbdec 100644
--- a/services/camera/libcameraservice/Android.mk
+++ b/services/camera/libcameraservice/Android.mk
@@ -66,6 +66,10 @@ ifeq ($(BOARD_HTC_3D_SUPPORT),true)
     LOCAL_CFLAGS += -DHTC_3D_SUPPORT
 endif
 
+ifeq ($(BOARD_CAMERA_NO_UNWANTED_MSG),true)
+    LOCAL_CFLAGS += -DCAMERA_NO_UNWANTED_MSG
+endif
+
 LOCAL_MODULE:= libcameraservice
 
 include $(BUILD_SHARED_LIBRARY)
diff --git a/services/camera/libcameraservice/CameraClient.cpp b/services/camera/libcameraservice/CameraClient.cpp
index 533a88c..331f60c 100644
--- a/services/camera/libcameraservice/CameraClient.cpp
+++ b/services/camera/libcameraservice/CameraClient.cpp
@@ -754,7 +754,9 @@ bool CameraClient::lockIfMessageWanted(int32_t msgType) {
         }
         usleep(CHECK_MESSAGE_INTERVAL * 1000);
     }
+#ifndef CAMERA_NO_UNWANTED_MSG
     ALOGW("lockIfMessageWanted(%d): dropped unwanted message", msgType);
+#endif
     return false;
 }
 
-- 
1.8.5.5
